R/functions.R

```diff
@@ -4001,23 +4001,43 @@ chmAddTSNE <- function(hm, axis, tsne, pointIds, basename = "TSNE") {
 #' @seealso [chmAddReducedDim()]
 
 chmAddPCA <- function(hm, axis, prc, basename = "PC", ndim = 2) {
-  stopifnot(is(hm, "ngchmVersion2"))
-  stopifnot(mode(axis) == "character" && length(axis) == 1)
-  stopifnot(axis == "row" || axis == "column")
-  stopifnot(mode(basename) == "character" && length(basename) == 1)
-  stopifnot(mode(ndim) == "numeric" && length(basename) == 1)
+  if (class(hm) != "ngchmVersion2") {
```

Comment:
Using is(object, "classname") is better than class(object)  != "classname" for two reasons:
- Inherited class objects
- Multiclassed objects (which will return a list of classes).
In both cases the test will fail even if the object belongs to an inherited class or is multi-classed. The is(object,"classname") method will work regardless.

Reply:
Need to use !is to be equivalent test.